proxy/src/serverless/http_conn_pool.rs

```diff
@@ -23,39 +25,43 @@ pub(crate) type Connect =
     http2::Connection<TokioIo<TcpStream>, hyper::body::Incoming, TokioExecutor>;
 
 #[derive(Clone)]
-pub(crate) struct ConnPoolEntry<C: ClientInnerExt + Clone> {
-    conn: C,
-    conn_id: uuid::Uuid,
-    aux: MetricsAuxInfo,
-}
-
 pub(crate) struct ClientDataHttp();
 
 // Per-endpoint connection pool
 // Number of open connections is limited by the `max_conns_per_endpoint`.
-pub(crate) struct EndpointConnPool<C: ClientInnerExt + Clone> {
+pub(crate) struct HttpConnPool<C: ClientInnerExt + Clone> {
     // TODO(conrad):
     // either we should open more connections depending on stream count
     // (not exposed by hyper, need our own counter)
     // or we can change this to an Option rather than a VecDeque.
     //
     // Opening more connections to the same db because we run out of streams
     // seems somewhat redundant though.
-    //
+    //E
     // Probably we should run a semaphore and just the single conn. TBD.
     conns: VecDeque<ConnPoolEntry<C>>,
     _guard: HttpEndpointPoolsGuard<'static>,
     global_connections_count: Arc<AtomicUsize>,
 }
 
-impl<C: ClientInnerExt + Clone> EndpointConnPool<C> {
+impl<C: ClientInnerExt + Clone> HttpConnPool<C> {
     fn get_conn_entry(&mut self) -> Option<ConnPoolEntry<C>> {
         let Self { conns, .. } = self;
 
         loop {
             let conn = conns.pop_front()?;
-            if !conn.conn.is_closed() {
-                conns.push_back(conn.clone());
+            if !conn.conn.inner.is_closed() {
+                let new_conn = ConnPoolEntry {
+                    conn: ClientInnerCommon {
+                        inner: conn.conn.inner.clone(),
+                        aux: conn.conn.aux.clone(),
+                        conn_id: conn.conn.conn_id,
+                        data: conn.conn.data.clone(),
+                    },
```

Comment:
Could this be:
```suggestion
                    conn: conn.conn.clone(),
```

Reply:
It seems, that it won't work w/out defining clone() for ClientInnerCommon. Before it was possible because of the different ConnPoolEntry

Reply:
@conradludgate do you want me to implement clone() for ClientInnerCommon or that's fine?